### ISSUE TITLE
[VL][1.2][Result mismatch] Cast string to integral type does not ignore ISO control characters

### ISSUE THREAD

**Issue by wForget** [apache/incubator-gluten#7749]
### Backend

VL (Velox)

### Bug description

1. create test dataset
    ```
    create table t1 as select url_decode('111111%00') as c1;
    ```

2. cast to integral type
    check sqls:
    ```
    select cast(c1 as int) from t1;
    select cast(c1 as bigint) from t1;
    ```
    gluten 1.2 + spark 3.5:
    ```
    NULL
    NULL
    ```
    vanilla spark:
    ```
    111111
    111111
    ```
3. cast to fractional type
    check sqls:
    ```
    select cast(c1 as float) from t1;
    select cast(c1 as double) from t1;
    ```
    gluten 1.2 + spark 3.5:
    ```
    111111.0
    111111.0
    ```
    vanilla spark:
    ```
    111111.0
    111111.0
    ```

Related PRs: https://github.com/apache/spark/pull/41535

### Spark version

None

**Comment by wForget** [comment]
I will try to fix it, it seems to be related to this code:

https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/specialforms/SparkCastHooks.cpp#L70-L75

**Comment by jackylee-ch** [comment]
String type conversions were also fixed in #1569 . They used trim to remove invalid characters, but that didn't handle invalid characters inside a String. It would be great to fix this in velox so we can remove #1569 in gluten.

**Comment by wForget** [comment]
> String type conversions were also fixed in #1569 . They used trim to remove invalid characters, but that didn't handle invalid characters inside a String. It would be great to fix this in velox so we can remove #1569 in gluten. cc @PHILO-HE

Thank you for the information. Since the trim behavior for different types of cast in spark is different, it seems that extending #1569 is a simpler solution, can we fix it that way first?
